crates/bevy_xpbd_2d/examples/one_way_platform_2d.rs

```diff
@@ -0,0 +1,305 @@
+/// set to disallow passing through.
+///
+/// > Note that this is a very simplistic implementation of one-way
+/// > platforms to demonstrate filtering collisions via [`PostProcessCollisionsSchedule`].
+/// > You will probably want something more robust to implement one-way
+/// > platforms properly, or may elect to use a sensor collider for your entities instead,
+/// > which means you won't need to filter collisions at all.
+///
+/// #### When an entity is known to already be passing through the [`OneWayPlatform`]
+/// Any time an entity begins passing through a [`OneWayPlatform`], it is added to the
+/// [`OneWayPlatform`]'s set of currently active penetrations, and will be allowed to
+/// continue to pass through the platform until it is no longer penetrating the platform.
+///
+/// The entity is allowed to continue to pass through the platform as long as at least
+/// one contact is penetrating.
+///
+/// Once all of the contacts are no longer penetrating the [`OneWayPlatform`], or all contacts
+/// have stopped, the entity is forgotten about and the logic falls through to the next part.
+///
+/// #### When an entity is NOT known to be passing through the [`OneWayPlatform`]
+/// Depending on the setting of [`PassThroughOneWayPlatform`], the entity may be allowed to
+/// pass through.
+///
+/// If no [`PassThroughOneWayPlatform`] is present, [`PassThroughOneWayPlatform::ByNormal`] is used.
+///
+/// [`PassThroughOneWayPlatform`] may be in one of three states:
+/// 1. [`PassThroughOneWayPlatform::ByNormal`]
+///     - This is the default state
+///     - The entity may be allowed to pass through the [`OneWayPlatform`] depending on the contact normal
+///         - If all contact normals are in line with the [`OneWayPlatform`]'s local-space up vector,
+///           the entity is allowed to pass through
+/// 2. [`PassThroughOneWayPlatform::Always`]
+///     - The entity will always pass through the [`OneWayPlatform`], regardless of contact normal
+///     - This is useful for allowing an entity to jump down through a platform
+/// 3. [`PassThroughOneWayPlatform::Never`]
+///     - The entity will never pass through the [`OneWayPlatform`], meaning the platform will act
+///       as normal hard collision for this entity
+///
+/// Even if an entity is changed to [`PassThroughOneWayPlatform::Never`], it will be allowed to pass
+/// through a [`OneWayPlatform`] if it is already penetrating the platform. Once it exits the platform,
+/// it will no longer be allowed to pass through.
+fn one_way_platform(
+    mut one_way_platforms_query: Query<&mut OneWayPlatform>,
+    other_colliders_query: Query<
+        Option<&PassThroughOneWayPlatform>,
+        (With<Collider>, Without<OneWayPlatform>), // NOTE: This precludes OneWayPlatform passing through a OneWayPlatform
+    >,
+    mut collisions: ResMut<Collisions>,
+) {
+    // This assumes that Collisions contains empty entries for entities
+    // that were once colliding but no longer are.
+
```

Comment:
Is this line break intended? Looks a bit out of place to me, but maybe it's just preference

```suggestion
```

Reply:
This was intentional, since it applies to the whole function but is less important than the full explanation, but if the extra empty line is distracting (it must be for you to have noticed it) then let's remove it.